cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -323,53 +348,81 @@ private void waitForExecutionTaskToFinish() {
       List<ExecutionTask> finishedTasks = new ArrayList<>();
       do {
         Cluster cluster = _metadataClient.refreshMetadata().cluster();
-        LOG.debug("Tasks in progress: {}", _executionTaskManager.inProgressTasks());
-        for (ExecutionTask task : _executionTaskManager.inProgressTasks()) {
+        LOG.debug("Tasks in execution: {}", _executionTaskManager.inExecutionTasks());
+        List<ExecutionTask> deadOrAbortingTasks = new ArrayList<>();
+        for (ExecutionTask task : _executionTaskManager.inExecutionTasks()) {
           TopicPartition tp = task.proposal.topicPartition();
           if (cluster.partition(tp) == null) {
+            // Handle topic deletion during the execution.
             LOG.debug("Task {} is marked as finished because the topic has been deleted", task);
             finishedTasks.add(task);
             _executionTaskManager.markTaskAborting(task);
             _executionTaskManager.markTaskDone(task);
-            continue;
-          }
-          boolean taskDone;
-          switch (task.proposal.balancingAction()) {
-            case REPLICA_MOVEMENT:
-              taskDone = isReplicaMovementDone(cluster, tp, task);
-                break;
-            case REPLICA_DELETION:
-              taskDone = isReplicaDeletionDone(cluster, tp, task);
-              break;
-            case REPLICA_ADDITION:
-              taskDone = isReplicaAdditionDone(cluster, tp, task);
-              break;
-            case LEADERSHIP_MOVEMENT:
-              taskDone = isLeadershipMovementDone(cluster, tp, task);
-              break;
-            default:
-              throw new IllegalStateException("Should never be here.");
-          }
-          if (taskDone) {
+          } else if (isTaskDone(cluster, tp, task)) {
+            // Check to see if the task is done.
             finishedTasks.add(task);
             _executionTaskManager.markTaskDone(task);
+          } else if (maybeMarkTaskAsDeadOrAborted(cluster, task, _stopRequested)) {
```

Comment:
Why do we pass the private variable of this class, `_stopRequested`, as an argument to the private method of this class `maybeMarkTaskAsDeadOrAborted`?